### DISCUSSION TITLE
Update Compose Configuration docs for UseDataClass

### DISCUSSION THREAD

**Discussion by drinkthestars** [detekt/detekt#5220] [Q&A] [answered]
### Issue
`UseDataClass` currently triggers for a common Compose class pattern that uses `mutableStateOf` `var`s and [`@Stable` annotation](https://github.com/android/compose-samples/blob/e6994123804b976083fa937d3f5bf926da4facc5/Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyState.kt):

```kotlin
@Stable
class FooState(
    val fooA: String,
    val fooB: Int
) {
    var fooC by mutableStateOf(false)
    var fooD by mutableStateOf<Float>(0f)
}
```

(see this common pattern also showcased in [android/compose-samples](https://github.com/android/compose-samples/blob/e6994123804b976083fa937d3f5bf926da4facc5/Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/survey/SurveyState.kt).)

### Suggestion
Update [`Configuration for Compose`](https://detekt.dev/docs/introduction/compose/) doc with a section for how to customize `UseDataClass`, maybe by suggesting `Set ignoreAnnotated to ['Stable']`, to ensure `UseDataClass` doesn't trigger for that pattern.

Let me know if this suggestion makes sense, happy to open a PR with suggested change!

**Comment by tinder-cesardiez** [comment]
That's what we did at Tinder. I wonder if the the main maintainers have an opinion about it.

**Comment by schalkms** [comment, marked as answer]
Maintainer here:
I’d say go for it. We always appreciate PRs from the community helping to form some best practices. This way, the detekt documentation improves incrementally.

**Comment by drinkthestars** [comment]
Great, thanks! Will have one up

**Comment by picklebento** [comment]
I would personally suggest that `@Immutable` might also be considered as `ignoreAnnotated` as it is a stricter form of `@Stable`.
